File: utopian-tree/utopian-tree.go
package utopiantree

func utopianTree(growthCycleMax int32) int32 {
	height := int32(1)

	for growthCycle := int32(1); growthCycle <= growthCycleMax; growthCycle++ {
		if growthCycle%2 == 0 {
			height++
		} else {
			height *= 2
		}
	}

	return height
}

File: utopian-tree/UtopianTree.java
public class UtopianTree {

  public static int utopianTree(int growthCycleMax) {
    int height = 1;

    for (int growthCycle = 1; growthCycle <= growthCycleMax; growthCycle++) {
      height = growthCycle % 2 == 0
        ? height + 1
        : height * 2;
    }

    return height;
  }

}
